test/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectVerifier.kt

```diff
@@ -13,41 +13,40 @@
  */
 package org.codice.compliance.verification.binding
 
-import com.sun.org.apache.xml.internal.resolver.readers.ExtendedXMLCatalogReader
-import org.apache.cxf.rs.security.saml.sso.SSOConstants.*
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.SAML_RESPONSE
 import org.codice.compliance.SAMLComplianceException
 import org.codice.compliance.SAMLComplianceExceptionMessage.*
 import org.codice.compliance.children
+import org.codice.compliance.saml.plugin.IdpRedirectResponse
 import org.codice.compliance.utils.TestCommon.Companion.ACS_URL
 import org.codice.compliance.utils.TestCommon.Companion.EXAMPLE_RELAY_STATE
 import org.codice.compliance.utils.TestCommon.Companion.idpMetadata
 import org.codice.security.sign.SimpleSign
 import org.codice.security.sign.SimpleSign.SignatureException.SigErrorCode
-import org.w3c.dom.Node
 import java.io.UnsupportedEncodingException
 import java.net.URLDecoder
 import java.nio.charset.StandardCharsets
 
-class RedirectVerifier(responseDom: Node, parsedResponse: Map<String, String>, givenRelayState: Boolean): BindingVerifier(responseDom, parsedResponse, givenRelayState) {
+class RedirectVerifier(val response: IdpRedirectResponse) : BindingVerifier() {
     /**
      * Verify the response for a redirect binding
      */
     override fun verifyBinding() {
-        verifyRequestParam(parsedResponse[SAML_RESPONSE])
-        verifyNoXMLSig(responseDom)
-        verifyRedirectRelayState(parsedResponse[RELAY_STATE], givenRelayState)
-        parsedResponse[SIGNATURE]?.let {
-            verifyRedirectSignature(it, parsedResponse[SAML_RESPONSE], parsedResponse[RELAY_STATE], parsedResponse[SIG_ALG])
-            verifyRedirectDestination(responseDom)
+        verifyRequestParam(response)
```

Comment:
✏️ Same comment as in `PostVerifier` regarding private methods and not passing the `response` param.